SMTP module segfaults

When using the `multiple` module and then assigning a group of targets by trigger to the `smtp` module, the latter segfaults regardless of the targets given.

# Reproduction

All reproductions are done in an Ubuntu 24.04 Docker container where Zgrab2 is installed via:

```
go install github.com/zmap/zgrab2/cmd/zgrab2@2b0b755f95b9b423b4eba0b11f5c991378947765
```

## Test - No Targets

```
# cat /local/zgrab2.config
[Application Options]
output-file="/dev/stdout"
input-file="/dev/stdin"
blocklist-file="/dev/null"

[smtp]
trigger="TCP-25"
name="TCP-25"
port=25
```

```
# zgrab2 multiple --config-file /local/zgrab2.config < /dev/null
INFO[0000] started grab at 2025-11-06T21:53:21Z         
INFO[0000] finished grab at 2025-11-06T21:53:21Z        
00h:00m:00s; Scan Complete; 0 targets scanned; 0.00 targets/sec; 0.0% success rate
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x55a6a0]

goroutine 1 [running]:
github.com/zmap/zgrab2/bin.ZGrab2Main()
	/root/go/pkg/mod/github.com/zmap/zgrab2@v1.0.0-!r!c2.0.20251103195550-2b0b755f95b9/bin/bin.go:202 +0x820
main.main()
	/root/go/pkg/mod/github.com/zmap/zgrab2@v1.0.0-!r!c2.0.20251103195550-2b0b755f95b9/cmd/zgrab2/main.go:11 +0x1c
```

## Test - Localhost Target

```
# cat /local/zgrab2.config
[Application Options]
output-file="/dev/stdout"
input-file="/dev/stdin"
blocklist-file="/dev/null"

[smtp]
trigger="TCP-25"
name="TCP-25"
port=25
```

```
# cat /local/zgrab2.targets
127.0.0.1, , TCP-25
```

```
# zgrab2 multiple --config-file /local/zgrab2.config < /local/zgrab2.targets 
INFO[0000] started grab at 2025-11-06T21:54:13Z         
{"ip":"127.0.0.1","data":{"TCP-25":{"status":"connection-timeout","protocol":"smtp","port":25,"timestamp":"2025-11-06T21:54:13Z","error":"dial context failed: dial tcp 127.0.0.1:25: connect: connection refused"}}}
INFO[0000] finished grab at 2025-11-06T21:54:13Z        
00h:00m:00s; Scan Complete; 1 targets scanned; 235.21 targets/sec; 0.0% success rate
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x55a6a0]

goroutine 1 [running]:
github.com/zmap/zgrab2/bin.ZGrab2Main()
	/root/go/pkg/mod/github.com/zmap/zgrab2@v1.0.0-!r!c2.0.20251103195550-2b0b755f95b9/bin/bin.go:202 +0x820
main.main()
	/root/go/pkg/mod/github.com/zmap/zgrab2@v1.0.0-!r!c2.0.20251103195550-2b0b755f95b9/cmd/zgrab2/main.go:11 +0x1c
```

--- a/bin/bin.go
+++ b/bin/bin.go
@@ -122,9 +122,9 @@ func ZGrab2Main() {
 		log.Fatalf("could not parse flags: %s", err)
 	}
 
+	var modTypes []string
 	if m, ok := flag.(*zgrab2.MultipleCommand); ok {
 		iniParser := zgrab2.NewIniParser()
-		var modTypes []string
 		var flagsReturned []any
 		if m.ConfigFileName == "-" {
 			modTypes, flagsReturned, err = iniParser.Parse(os.Stdin)
@@ -159,6 +159,7 @@ func ZGrab2Main() {
 			zgrab2.RegisterScan(s.GetName(), s)
 		}
 	} else {
+		modTypes = append(modTypes, moduleType)
 		mod := zgrab2.GetModule(moduleType)
 		s := mod.NewScanner()
 		scanModuleNameToScanModule[moduleType] = &s
@@ -169,7 +170,7 @@ func ZGrab2Main() {
 	}
 	zgrab2.ValidateAndHandleFrameworkConfiguration() // will panic if there is an error
 	wg := sync.WaitGroup{}
-	monitor := zgrab2.MakeMonitor(1, &wg)
+	monitor := zgrab2.MakeMonitor(1, &wg, modTypes)
 	monitor.Callback = func(_ string) {
 		dumpHeapProfile()
 	}
@@ -199,7 +200,6 @@ func ZGrab2Main() {
 		metadata := (*module).GetScanMetadata()
 		if metadata != nil {
 			// Need to lookup the appropriate field in the summary
-			s.PerModuleMetadata[moduleName].CustomMetadata = metadata
 			moduleSummaryMetadata, ok := s.PerModuleMetadata[moduleName]
 			if ok {
 				moduleSummaryMetadata.CustomMetadata = metadata

--- a/monitor.go
+++ b/monitor.go
@@ -90,11 +90,15 @@ func (m *Monitor) printStatus(isFinalPrint bool) {
 
 // MakeMonitor returns a Monitor object that can be used to collect and send
 // the status of a running scan
-func MakeMonitor(statusChanSize int, wg *sync.WaitGroup) *Monitor {
+func MakeMonitor(statusChanSize int, wg *sync.WaitGroup, moduleNames []string) *Monitor {
 	m := &Monitor{
 		statusesChan: make(chan moduleStatus, statusChanSize),
 		states:       make(map[string]*ModuleMetadata),
 		startTime:    time.Now(),
+	}
+	// Initialize an empty state for each module
+	for _, moduleName := range moduleNames {
+		m.states[moduleName] = &ModuleMetadata{}
 	}
 	wg.Add(1)
 	go func() {

